From 6934249e4af208748d2c85a757470187a1d8fae2 Mon Sep 17 00:00:00 2001
From: Vincent Ahlrichs <vincent.ahlrichs@aisec.fraunhofer.de>
Date: Wed, 20 Jul 2022 13:58:07 +0200
Subject: [PATCH 1/2] riscv/vdso-cheri: Fix location _vdso_data symbol

The vdso data is placed on the previous page of the vdso. This is in
accordance with the non-cheri vdso
---
 arch/riscv/kernel/vdso-cheri/vdso.lds.S | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/riscv/kernel/vdso-cheri/vdso.lds.S b/arch/riscv/kernel/vdso-cheri/vdso.lds.S
index e6f558bca71b..e9111f700af0 100644
--- a/arch/riscv/kernel/vdso-cheri/vdso.lds.S
+++ b/arch/riscv/kernel/vdso-cheri/vdso.lds.S
@@ -3,12 +3,13 @@
  * Copyright (C) 2012 Regents of the University of California
  */
 #include <asm/page.h>
+#include <asm/vdso.h>
 
 OUTPUT_ARCH(riscv)
 
 SECTIONS
 {
-	PROVIDE(_vdso_data = . + PAGE_SIZE);
+	PROVIDE(_vdso_data = . - __VVAR_PAGES * PAGE_SIZE);
 	. = SIZEOF_HEADERS;
 
 	.hash		: { *(.hash) }			:text
-- 
2.37.3

